atuin-client/src/database.rs

```diff
@@ -52,6 +53,34 @@ pub struct Sqlite {
     pool: SqlitePool,
 }
 
+
+fn reorder_fuzzy(mode: SearchMode, query: &str, res: Vec<History>) -> Vec<History> {
+    match mode {
+	SearchMode::ReorderedFuzzy => {
+	    reorder(query, |x| &x.command, res)
+	},
+	_ => res,
+    }
+}
+
+
+fn reorder<F,A>(query: &str, f: F, res: Vec<A>) -> Vec<A>
```

Comment:
Could this be broken into another file please? In the future I'd like to go ahead with making some much more intelligent + configurable sorting, so it would be a nice start!